src/test/java/org/broadinstitute/hellbender/engine/dataflow/datasources/VariantShardTest.java

```diff
@@ -0,0 +1,40 @@
+package org.broadinstitute.hellbender.engine.dataflow.datasources;
+
+import org.broadinstitute.hellbender.utils.read.ArtificialReadUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+public final class VariantShardTest {
+
+    @DataProvider(name = "variantShards")
+    public Object[][] variantShards(){
+        List<GATKRead> reads = Arrays.asList(makeRead(1, 300, 1), makeRead(100000, 10, 2), makeRead(299999, 2, 3));
```

Comment:
Same comments as above about `DataProvider` organization and the dependence on the exact size of the shards.